## JOBSHEET 6

## PEMILIHAN 2

### Tujuan

Mahasiswa memahami tentang operator logika; Mahasiswa mampu menyelesaikan permasalahan dengan menggunakan sintaks pemilihan bersarang; Mahasiswa mampu membuat sebuah program Java yang memanfaatkan sintaks pemilihan bersarang


### Alat dan Bahan
+ PC/laptop
+ Browser(chrome, firefox, safari)
+ Koneksi internet

### Praktikum

#### Percobaan 1

#### Waktu percobaan : 40 menit

1. Tambahkan library Scanner, deklarasi Scanner

2. Buatlah variabel nilai yang memiliki tipe data int untuk menampung data yang diinput melalui keyboard

    ![](images/03.png4.png)

In [6]:
// Ketik kode di sini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
int nilai;
System.out.print("Masukan nilai : ");
nilai = sc.nextInt();

Masukan nilai : 105


3. Buatlah struktur pengecekan kondisi bersarang. Pengecekan pertama digunakan untuk memastikan bahwa nilai yang dimasukkan berada pada rentang 0 – 100. Jika nilai berada pada rentang 0 – 100, maka akan dilakukan pengecekan status kelulusan mahasiswa, yaitu jika nilai di antara 90 – 100 maka nilainya A, jika nilai di antara 80 – 89 maka nilainya B, jika nilai di antara 60 – 79 maka nilainya C, jika nilai di antara 50 – 59 maka nilainya D, dan jika nilai di antara 0 – 49 maka nilainya E. Sedangkan jika nilai berada di luar rentang 0 – 100, maka ditampilkan informasi bahwa nilai yang dimasukkan tidak valid.

    ![](images/04.png)

In [2]:
// Ketik kode di sini
if(nilai >= 0 && nilai <= 100){
    if(nilai >= 90 && nilai <=100){
        System.out.println("Selamat Anda mendapat nilai A");
    }else if (nilai >= 80 && nilai <= 89){
        System.out.println("Selamat Anda mendapat nilai B");
    }else if (nilai >= 60 && nilai <= 79){
        System.out.println("Selamat Anda mendapat nilai C");
    }else if (nilai >= 50 && nilai <= 59){
        System.out.println("Selamat Anda mendapat nilai D");
    }else{
        System.out.println("Selamat Anda mendapat nilai E");
    }
}else{
    System.out.println("Nilai yang Anda masukan tidak valid");
}

Selamat Anda mendapat nilai E


> Penjelasan kode program percobaan 1
> Pada program diatas pertama akan mengecek inputan nilai berada pada rentang 0 - 100, jika benar maka akan mengecek nilai sesuai dengan rentang status nilai.

##### Pertanyaan

1. Modifikasi kode program pada Percobaan 1 sehingga jika nilai yang dimasukkan kurang dari 0 akan ditampilkan output “Nilai yang Anda masukkan kurang dari 0” dan jika nilai yang dimasukkan lebih dari 100 akan ditampilkan output “Nilai yang Anda masukkan lebih dari 100”!

2. Jelaskan fungsi sintaks if (nilai >= 0 && nilai <= 100)!

3. Ubah operator && menjadi || pada sintaks if (nilai >= 0 && nilai <= 100). Jalankan program dengan memasukkan nilai = 105. Amati apa yang terjadi! Mengapa hasilnya demikian?

In [3]:
// Jawaban pertanyaan
// Ketik kode di sini
if(nilai >= 0 && nilai <= 100){
    if(nilai >= 90 && nilai <=100){
        System.out.println("Selamat Anda mendapat nilai A");
    }else if (nilai >= 80 && nilai <= 89){
        System.out.println("Selamat Anda mendapat nilai B");
    }else if (nilai >= 60 && nilai <= 79){
        System.out.println("Selamat Anda mendapat nilai C");
    }else if (nilai >= 50 && nilai <= 59){
        System.out.println("Selamat Anda mendapat nilai D");
    }else{
        System.out.println("Selamat Anda mendapat nilai E");
    }
}else if(nilai > 100){
    System.out.println("Nilai yang Anda masukkan lebih dari 100");
}else if(nilai < 0){
    System.out.println("Nilai yang Anda masukkan kurang dari 0");
}
else{
    System.out.println("Nilai yang Anda masukan tidak valid");
}

Selamat Anda mendapat nilai E


> 2. Digunakan untuk mengecek kondisi jika nilai lebih dari sama dengan 0 dan nilai kurang dari sama dengan 100
> 3. Perkondisi tersebut akan menghasilkan benar karena salah satu kondisi terpenuhi yaitu nilai lebih dari sama 0 walaupun nilai kurang dari sama dengan 100 tidak terpenuhi karena menggunakan operator OR maka akan dinilai benar jika salah satu benar.

In [7]:
// Jawaban pertanyaan
// Ketik kode di sini
if(nilai >= 0 || nilai <= 100){
    if(nilai >= 90 && nilai <=100){
        System.out.println("Selamat Anda mendapat nilai A");
    }else if (nilai >= 80 && nilai <= 89){
        System.out.println("Selamat Anda mendapat nilai B");
    }else if (nilai >= 60 && nilai <= 79){
        System.out.println("Selamat Anda mendapat nilai C");
    }else if (nilai >= 50 && nilai <= 59){
        System.out.println("Selamat Anda mendapat nilai D");
    }else{
        System.out.println("Selamat Anda mendapat nilai E");
    }
}else if(nilai > 100){
    System.out.println("Nilai yang Anda masukkan lebih dari 100");
}else if(nilai < 0){
    System.out.println("Nilai yang Anda masukkan kurang dari 0");
}
else{
    System.out.println("Nilai yang Anda masukan tidak valid");
}

Selamat Anda mendapat nilai E


#### Percobaan 2

#### Waktu percobaan : 40 menit

1. Perhatikan flowchart dibawah ini!

![](images/02.png)

> Flowchart tersebut digunakan untuk menghitung gaji bersih seseorang setelah dipotong pajak sesuai dengan kategorinya (pekerja dan pebisnis) dan besarnya penghasilan. 

2. Tambahkan library Scanner dan deklarasi Scanner

3. Deklarasikan variabel kategori, penghasilan, gajiBersih, dan pajak

    ![](images/05.png)

In [6]:
// Ketik kode di sini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
String kategori;
int penghasilan,gajiBersih;
double pajak;

System.out.print("Masukkan kategori : ");
kategori = sc.nextLine();
System.out.print("Masukkan penghasilan :");
penghasilan = sc.nextInt();

Masukkan kategori : PEBISNIS
Masukkan penghasilan :100000


4. Buatlah struktur pengecekan kondisi bersarang. Pengecekan pertama digunakan untuk mengecek kategori (pekerja atau pebisnis). Selanjutnya dilakukan pengecekan kedua untuk menentukan besarnya pajak berdasarkan penghasilan yang telah dimasukkan.Kemudian tambahkan kode program untuk menghitung gaji bersih yang diterima setelah dipotong pajak!

    ![](images/06.png)

In [7]:
// Ketik kode di sini
if(kategori.equals("pekerja")){
    if(penghasilan <= 2000000){
        pajak = 0.1;
    }
    else if (penghasilan <= 3000000){
        pajak = 0.15;
    }else{
        pajak =0.2;
    }
    gajiBersih = (int) (penghasilan - (penghasilan * pajak));
    System.out.println("Gaji bersih anda segini : " + gajiBersih);
}else if(kategori.equals("pebisnis")){
     if(penghasilan <= 2500000){
        pajak = 0.15;
    }
    else if (penghasilan <= 3500000){
        pajak = 0.2;
    }else{
        pajak =0.25;
    }
    gajiBersih = (int) (penghasilan - (penghasilan * pajak));
    System.out.println("Gaji bersih anda segini : " + gajiBersih);
}
else{
    System.out.println("Kategori Salah");
}


Kategori Salah


5. Jalankan program di atas. Amati apa yang terjadi!

> Penjelasan kode program percobaan 2
> Program diatas digunakan untuk menghitung gaji bersih pekerja atau pebisnis. pertama user menginputkan kategori pekerjaan dan penghasilan lalu akan di cek jika kategori pekerja atau pebisnis setelah itu lanjut ke pengecheckan gaji masing masing kategori lalu pajak akan sesuai dengan rentang gaji. setelah itu dihitung deh gaji bersihnya dan dikeluarkan gaji bersihnya.

##### Pertanyaan

1. Jalankan program dengan memasukkan kategori = pekerja dan penghasilan = 2048485. Amati apa yang terjadi! Mengapa angka di belakang koma tidak ditampilkan?

2. Jelaskan fungsi dari (int) pada sintaks:
```
gajiBersih = (int) (penghasilan - (penghasilan * pajak));
```

3.	Jalankan program dengan memasukkan kategori = PEBISNIS dan penghasilan = 2000000. Amati apa yang terjadi! Apa kegunaan dari equalsIgnoreCase?

4.	Ubah equalsIgnoreCase menjadi equals, kemudian jalankan program dengan memasukkan kategori = PEBISNIS dan penghasilan = 2000000. Amati apa yang terjadi! Mengapa hasilnya demikian? Apa kegunaan dari equals?

> 1. karena tipe data gajiBersih adalah int sehingga decimalnya tidak ditampilkan bedalagi kalau pakai double atau float.
> 2. fungsi diatas untuk menghitung gajiBersih lalu hasilnya dilakukan narrowing casting ke integer.
> 3. equalsIgnoreCase berguna untuk membandingkan String apakah sama atau tidak dengan mengecualikan case nya jadi mau pake huruf besar atau kecil sama saja yang penting kalimatnya sama.
> 4. karena menggunakan equals saja maka jika casenya tidak sama maka akan salah. Kegunaan equals untuk membandingkan String apakah sama atau tidak tetapi masih memperhatikan casenya.

In [7]:
// Jawaban pertanyaan

### Tugas

#### Waktu pengerjaan Tugas: 140 menit

1. Buatlah program kalkulator sederhana menggunakan bahasa pemrograman Java. User akan menginputkan dua buah bilangan riil dan satu buah operator aritmatika (+, -, *, atau /), kemudian program akan mengoperasikan dua bilangan tersebut dengan operator yang sesuai. Petunjuk: gunakan pernyataan switch-case.
Contoh tampilan program:

```
Masukkan bilangan pertama: 2.5
Masukkan operator (+, -, *, /): *
Masukkan bilangan kedua: 4
2.5 * 4.0 = 10.0

```

In [8]:
//ketik kode program dan lampirkan hasilnya disini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
char operator;
double angka1,angka2,hasil;

// Ketik kode program di atas di bawah sini
System.out.print("Masukkan angka pertama : ");
angka1 = sc.nextDouble();
System.out.print("Masukkan operator (+ - * /) : ");
operator = sc.next().charAt(0);
System.out.print("Masukkan angka kedua : ");
angka2 = sc.nextDouble();

switch(operator){
    case '+':
        hasil = angka1 + angka2;
        System.out.println(angka1 + " + " + angka2 + " = " + hasil);
        break;
    case '-':
        hasil = angka1 - angka2;
        System.out.println(angka1 + " - " + angka2 + " = " + hasil);
        break;
    case '*':
        hasil = angka1 * angka2;
        System.out.println(angka1 + " * " + angka2 + " = " + hasil);
        break;
    case '/':
        hasil = angka1 / angka2;
        System.out.println(angka1 + " / " + angka2 + " = " + hasil);
        break;
    default:
        System.out.println("Operator yang Anda masukkan salah");
}

Masukkan angka pertama : 20
Masukkan operator (+ - * /) : 20
Masukkan angka kedua : 2130
Operator yang Anda masukkan salah


2. Dengan menggunakan tiga nilai yang mewakili panjang tiga sisi sebuah segitiga, tentukan apakah segitiga tersebut sama sisi (ketiga sisinya bernilai sama), sama kaki (kedua sisinya bernilai sama), atau sembarang (tidak ada sisi yang bernilai sama)! 

In [9]:
//ketik kode program dan lampirkan hasilnya disini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
double sisi1,sisi2,sisi3;

System.out.print("Masukkan panjang sisi 1 : ");
sisi1 = sc.nextDouble();
System.out.print("Masukkan panjang sisi 2 : ");
sisi2= sc.nextDouble();
System.out.print("Masukkan panjang sisi 3 : ");
sisi3= sc.nextDouble();

if(sisi1 == sisi2 && sisi1 == sisi3 ){
    System.out.println("ini adalah segitiga sama sisi");
}else if((sisi1 == sisi2) || (sisi2 == sisi3) || sisi3 == sisi1){
    System.out.println("ini adalah segitiga sama kaki");
}else{
    System.out.println("ini adlah segitiga sembarang");
}

Masukkan panjang sisi 1 : 2130
Masukkan panjang sisi 2 : 1230
Masukkan panjang sisi 3 : 1230
ini adalah segitiga sama kaki


3. Warung Padang Gembira meminta Anda membuat sebuah program untuk menerima pesanan dari internet. Program yang Anda buat meminta user untuk memasukkan nama makanan dan harga. Setelah itu, user ditawarkan untuk menggunakan pengiriman ekspres. Jika pengguna menolak, maka jenis pengiriman yang digunakan adalah pengiriman reguler. Biaya pengiriman reguler untuk harga makanan kurang dari Rp 100.000 adalah Rp 20.000, sedangkan untuk harga makanan sama dengan atau lebih dari Rp 100.000 biaya pengirimannya adalah Rp 30.000. Untuk jenis pengiriman ekspres, tambahkan biaya tambahan sebesar Rp 25.000 dari standar biaya pengiriman reguler. Tampilkan struk yang berisi nama makanan yang dibeli + harga, biaya pengiriman, dan total yang harus dibayar!
Contoh hasil output program:

```
Masukkan nama makanan: Tuna salad
Masukkan harga makanan: Rp 115000
Apakah Anda ingin pengiriman ekspres (0 = tidak, 1 = ya)? 0

STRUK PEMBELIAN
Tuna salad        Rp 115000
Biaya pengiriman  Rp 30000
TOTAL             Rp 145000

```

```
Masukkan nama makanan: Beef bulgogi
Masukkan harga makanan: Rp 78000
Apakah Anda ingin pengiriman ekspres (0 = tidak, 1 = ya)? 1

STRUK PEMBELIAN
Beef bulgogi      Rp 115000
Biaya pengiriman  Rp 45000
TOTAL             Rp 123000

```



In [10]:
//ketik kode program dan lampirkan hasilnya disini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
String namaMakanan;
long hargaMakanan,biayaPengiriman, total;
byte isEkspres;

System.out.print("Masukkan nama makanan : ");
namaMakanan = sc.nextLine();
System.out.print("Masukkan harga makanan : ");
hargaMakanan = sc.nextLong();
System.out.print("Apakah Anda ingin pengiriman exspres (0 = tidak, 1 = ya)?");
isEkspres = sc.nextByte();

if(hargaMakanan >= 100000)
    biayaPengiriman = 30000;
else
    biayaPengiriman = 20000;

if(isEkspres == 1)
    biayaPengiriman +=25000;

total = hargaMakanan + biayaPengiriman;

System.out.println("STRUK PEMBELIAN");
System.out.println(namaMakanan + "\t\t Rp " + hargaMakanan);
System.out.println("Biaya Pengiriman \t Rp " + biayaPengiriman);
System.out.println("Total \t\t\t Rp " + total);

Masukkan nama makanan : 1023
Masukkan harga makanan : 1230
Apakah Anda ingin pengiriman exspres (0 = tidak, 1 = ya)?1203


EvalException: Value out of range. Value:"1203" Radix:10

4. Perhatikan flowchart berikut ini!

![](images/01.png)

> Buatlah program sesuai dengan flowchart diatas!

In [1]:
//ketik kode program dan lampirkan hasilnya disini
import java.util.Scanner;

Scanner sc = new Scanner(System.in);
int umur,tanggungan;
boolean isMiskin;
double pendapatan,biayaHidup;

System.out.print("Masukkan umur Anda : ");
umur = sc.nextInt();

if(umur >= 18){
    System.out.print("Sudah bekerja ? (sudah atau belum) : ");
    String statusBekerja = sc.next();
    if(statusBekerja.equalsIgnoreCase("sudah")){
        System.out.print("Masukkan pendapatan perbulan anda : ");
        pendapatan = sc.nextDouble();
        System.out.print("Masukkan jumlah tanggungan Anda : ");
        tanggungan = sc.nextInt();
        biayaHidup = pendapatan / tanggungan;
        if(biayaHidup < 300000)
            isMiskin = true;
    }else if(statusBekerja.equalsIgnoreCase("belum")){
        isMiskin = true;
    }else{
        System.out.print("Maaf inputan yang anda masukan tidak valid");
        System.exit(0);
    }
}else{
    System.out.print("Masih sekolah ? (ya atau tidak) : ");
    String statusSekolah = sc.next();
    if(statusSekolah.equalsIgnoreCase("ya"))
       isMiskin = false;
    else if(statusSekolah.equalsIgnoreCase("tidak"))
        isMiskin = true;
    else {
        System.out.println("Maaf inputan yang anda masukan tidak valid");
        System.exit(0);
    }
}

if(isMiskin)
    System.out.println("Anda termasuk penduduk miskin");
else
    System.out.println("Anda bukan termasuk penduduk miskin");

Masukkan umur Anda : 18
Sudah bekerja ? (sudah atau belum) : sudah
Masukkan pendapatan perbulan anda : 100000
Masukkan jumlah tanggungan Anda : 2
Anda termasuk penduduk miskin
